eng/Microsoft.Extensions.targets

```diff
@@ -3,19 +3,7 @@
     <_UseNet6Packages Condition=" '$(TargetFramework.Contains(net6.0))' == 'true' or '$(PackageType)' == 'DotnetPlatform' ">true</_UseNet6Packages>
   </PropertyGroup>
   <PropertyGroup Condition=" '$(_UseNet6Packages)' != 'true' ">
-    <_MicrosoftHostingVersion>5.0.0</_MicrosoftHostingVersion>
```

Comment:
I dropped all these versions because they don't seem to be needed. All the 6.0.0 versions of these packages continue to support the same TFMs as before, so I don't think there's any need to restrict parts of the build to the 5.0.0 versions. The _real_ reason I removed it is to avoid some build issue where diamond dependencies were causing NuGet errors. Either way, not needed as far as I know.